packages/packages/site-settings/src/components/__tests__/ported-primary-action.test.tsx

```diff
@@ -0,0 +1,93 @@
+import { render, screen } from '@testing-library/react';
+import { isRouteActive } from '@elementor/v1-adapters';
+import { useActiveDocument } from '@elementor/documents';
+import PortedPrimaryAction from '../ported-primary-action';
+import { createMockDocument } from 'test-utils';
+
+jest.mock( '@elementor/v1-adapters', () => ( {
+	...jest.requireActual( '@elementor/v1-adapters' ),
+	isRouteActive: jest.fn(),
+} ) );
+
+jest.mock( '@elementor/documents', () => ( {
+	useActiveDocument: jest.fn(),
+	useActiveDocumentActions: jest.fn( () => ( {
+		save: jest.fn(),
+	} ) ),
+} ) );
+
+describe( '@elementor/documents - Site settings ported primary action', () => {
+	beforeEach( () => {
+		jest.mocked( useActiveDocument ).mockImplementation( () => createMockDocument() );
+	} );
+
+	afterEach( () => {
+		jest.clearAllMocks();
+
+		document.body.innerHTML = '';
+	} );
+
+	it( 'should render the button when navigating to site-settings route', () => {
+		// Arrange.
+		createRootElement();
+
+		render( <PortedPrimaryAction /> );
+
+		// Assert.
+		expect( screen.queryByRole( 'button' ) ).not.toBeInTheDocument();
+
+		// Act.
+		jest.mocked( isRouteActive ).mockImplementation( () => true );
+
+		navigateTo( 'panel/global/menu' );
+
+		// Assert.
+		expect( screen.queryByRole( 'button' ) ).toBeInTheDocument();
+	} );
+
+	it( 'should not render button when navigation to another route', () => {
+		// Arrange.
+		createRootElement();
+
+		jest.mocked( isRouteActive ).mockImplementation( () => false );
+
+		render( <PortedPrimaryAction /> );
+
+		// Assert.
+		expect( screen.queryByRole( 'button' ) ).not.toBeInTheDocument();
+
+		// Act.
+		jest.mocked( isRouteActive ).mockImplementation( () => true );
+
+		navigateTo( 'panel/history' );
+
+		// Assert.
+		expect( screen.queryByRole( 'button' ) ).not.toBeInTheDocument();
+	} );
+
+	it( 'should not render button when root element not exists', () => {
+		// Arrange.
+		createRootElement( 'some-fake-id' );
+
+		jest.mocked( isRouteActive ).mockImplementation( () => true );
+
+		render( <PortedPrimaryAction /> );
+
+		// Assert.
+		expect( screen.queryByRole( 'button' ) ).not.toBeInTheDocument();
+	} );
+} );
+
+function createRootElement( id = 'elementor-panel-inner' ) {
+	const el = document.createElement( 'div' );
+
+	el.setAttribute( 'id', id );
+
+	document.body.appendChild( el );
+}
+
+function navigateTo( route: string ) {
+	window.dispatchEvent( new CustomEvent( 'elementor/routes/open', {
+		detail: { route },
+	} ) );
+}
```

Comment:
maybe we need to move it to the test utils, and use it everywhere, and also the command utils I created

Reply:
For another task